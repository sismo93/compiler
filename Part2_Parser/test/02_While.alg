begin
  while b /= 0 do
    b := a ;
    a := c
  endwhile 
end
