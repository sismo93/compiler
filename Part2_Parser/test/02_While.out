Rules used : 1 3 4 9 45 31 34 37 14 15 17 23 21 44 14 15 18 23 21 36 33 3 4 7 13 14 15 17 23 21 5 4 7 13 14 15 17 23 21 6 6 
 Here are all the grammar used with more detail : 

<Program> -> begin <Code> end

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <While>

<while> -> while <cond> do <code> endwhile

<Cond> -> <andCond> <condPrim>

<andCond> -> <simpleCond> <andCondPrim>

<simpleCond> -> <exprArith> <comp> <exprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<comp> -> /=

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> NUMBER

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<andCondPrim> -> EPSILON

<condPrim> -> EPSILON

<Code> -> <InstList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Assign>

<Assign> -> [VarName] := <ExprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<InstListPrim> -> SEMICOLON <instList>

<InstList> -> <Instruction> <InstListPrim>

<Instruction> -> <Assign>

<Assign> -> [VarName] := <ExprArith>

<ExprArith> -> <Term> <ExprArith_prim>

<Term> -> <Atom> <prodPrim>

<Atom> -> VARNAME

<prodPrim> -> EPSILON

<ExprArith_prim> -> EPSILON

<InstListPrim> -> EPSILON

<InstListPrim> -> EPSILON
